mws/mws.go

```diff
@@ -46,19 +46,30 @@ type NetworkVPCEndpoints struct {
 	DataplaneRelayAPI []string `json:"dataplane_relay" tf:"slice_set"`
 }
 
+// GcpNetworkInfo is the object that configures byovpc settings for gcp
+type GcpNetworkInfo struct {
+	NetworkProjectId   string `json:"network_project_id"`
+	VpcId              string `json:"vpc_id"`
+	SubnetId           string `json:"subnet_id"`
+	SubnetRegion       string `json:"subnet_region"`
+	PodIpRangeName     string `json:"pod_ip_range_name"`
+	ServiceIpRangeName string `json:"service_ip_range_name"`
+}
+
 // Network is the object that contains all the information for BYOVPC
 type Network struct {
 	AccountID        string               `json:"account_id"`
 	NetworkID        string               `json:"network_id,omitempty" tf:"computed"`
 	NetworkName      string               `json:"network_name"`
-	VPCID            string               `json:"vpc_id"`
-	SubnetIds        []string             `json:"subnet_ids" tf:"slice_set"`
+	VPCID            string               `json:"vpc_id,omitempty"`
+	SubnetIds        []string             `json:"subnet_ids,omitempty" tf:"slice_set"`
 	VPCEndpoints     *NetworkVPCEndpoints `json:"vpc_endpoints,omitempty" tf:"computed,force_new"`
-	SecurityGroupIds []string             `json:"security_group_ids" tf:"slice_set"`
+	SecurityGroupIds []string             `json:"security_group_ids,omitempty" tf:"slice_set"`
 	VPCStatus        string               `json:"vpc_status,omitempty" tf:"computed"`
 	ErrorMessages    []NetworkHealth      `json:"error_messages,omitempty" tf:"computed"`
 	WorkspaceID      int64                `json:"workspace_id,omitempty" tf:"computed"`
 	CreationTime     int64                `json:"creation_time,omitempty" tf:"computed"`
+	GcpNetworkInfo   *GcpNetworkInfo      `json:"gcp_network_info,omitempty"`
```

Comment:
If you don’t specify this on gcp, what read operation returns? Maybe we need suppress diff?

Reply:
afaik you cannot create a network object in gcp without this field being configured. I will be testing it on aws which shouldnt return an object for this struct. @bhavink-db can correct me if i am wrong.

Reply:
I meant that sometimes get api returns less information, or instead of null returns empty object

Reply:
i validated this via drift to see if anything changes; after creating it i ran apply again to avoid any drift

but i can do two additional things:

1. confirm with eng on this api to confirm that the input is exactly what will be returned by the get and not be omitted or returned as a null. 
2. I think i can also confirm this by trying an import for an existing network and see if there is any drift.

or if you think we should add a suppress diff

@alexott let me know if this is sufficient.